lib/project_types/script/script_project.rb

```diff
@@ -2,47 +2,43 @@
 
 module Script
   class ScriptProject < ShopifyCli::Project
-    SUPPORTED_LANGUAGES = %w(ts)
-    SOURCE_DIR = "src"
-
     attr_reader :extension_point_type, :script_name, :language
 
     def initialize(*args)
       super
-      @extension_point_type = lookup_config('extension_point_type')
+      @extension_point_type = lookup_config!('extension_point_type')
       raise Errors::DeprecatedEPError, @extension_point_type if deprecated?(@extension_point_type)
-      @script_name = lookup_config('script_name')
-      @language = 'ts'
+      @script_name = lookup_config!('script_name')
+      @language = lookup_language
       ShopifyCli::Core::Monorail.metadata = {
         "script_name" => @script_name,
         "extension_point_type" => @extension_point_type,
         "language" => @language,
       }
     end
 
-    def file_name
-      "script.#{language}"
-    end
-
-    def source_file
-      "#{SOURCE_DIR}/#{file_name}"
-    end
-
-    def source_path
-      "#{script_name}/#{source_file}"
-    end
-
     private
 
     def deprecated?(ep)
       Script::Layers::Application::ExtensionPoints.deprecated_types.include?(ep)
     end
 
     def lookup_config(key)
+      return nil unless config.key?(key)
+      config[key]
+    end
+
+    def lookup_config!(key)
       raise Errors::InvalidContextError, key unless config.key?(key)
       config[key]
     end
 
+    def lookup_language
+      lang = lookup_config('language') || 'AssemblyScript'
```

Comment:
We have to have this fallback here because we were not storing the current language anywhere and we don't want to break existing scripts

Reply:
Right, so we are storing the language from this change on right? I think that the assumption that if there's no language defined the current script project is using AS is the right call.